node/src/actors/chain_manager/handlers.rs

```diff
@@ -511,12 +524,16 @@ impl Handler<PeersBeacons> for ChainManager {
         // The beacons are Option<CheckpointBeacon>, so peers that have not
         // sent us a beacon are counted as None. Keeping that in mind, we
         // reach consensus as long as consensus_threshold % of peers agree.
-        // In case of tie returns None
-        let consensus = mode_consensus(pb.iter().map(|(_p, b)| b), consensus_threshold)
-            // Flatten result:
-            // None (no consensus) should be treated the same way as
-            // Some(None) (the consensus is that there is no consensus)
-            .and_then(|x| *x);
+        let consensus = mode_consensus(
+            pb.iter()
+                .map(|(_p, b)| b)
+                .chain(std::iter::repeat(&None).take(num_missing_peers)),
+            consensus_threshold,
+        )
+        // Flatten result:
+        // None (consensus % below threshold) should be treated the same way as
+        // Some(None) (most of the peers did not send a beacon)
```

Comment:
What's this commenting on?

Reply:
The and_then below, which maps Some(None) to None

Reply:
My bad!